### PR TITLE
Add codeowners for PR automation

### PR DESCRIPTION
**Pull request by rossf7**
#### What type of PR is this?

kind/enhancement

#### What this PR does / why we need it:

Adds CODEOWNERS file to automate setting of reviewers in PRs

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer (optional):

Hey @leonardpahlke @guidemetothemoon,
we would like to add this automation so its clear who should review PRs and so we can spread the reviews across the team.

Are you OK with this?

I'm also going to need some help to enable it as I don't have access to the branch protection settings.

### CODE REVIEW

**Comment by AntonioDiTuri** [comment]
Looks good to me for the moment, maybe we can specify some rules later on :)

**Review by AntonioDiTuri — APPROVED** [review]

**Comment by AntonioDiTuri** [comment]
I am waiting for your ack to merge  @leonardpahlke, @guidemetothemoon if it is ok to be listed as reviewers :)

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+* @nikimanoledaki 
+* @AntonioDiTuri
+* @rossf7
+* @leonardpahlke
+* @guidemetothemoon
```
Comment (leonardpahlke) [inline CODEOWNERS:5]:
```suggestion 
* @cncf-tags/tag-env-leads 
* @cncf-tags/tag-env-wg-green-reviews-leads

LICENSE @cncf-tags/tag-env-leads 
*.yaml @cncf-tags/tag-env-wg-green-reviews-leads
*.tf @cncf-tags/tag-env-wg-green-reviews-leads
/docs/* @cncf-tags/tag-env-wg-green-reviews-leads
```

**Review by leonardpahlke — COMMENTED** [review]
lets use github groups and specify some files & folders

**Comment by rossf7** [comment]
@leonardpahlke Thank you, better to use the groups.

**Review by leonardpahlke — APPROVED** [review]
LGTM

**Comment by leonardpahlke** [comment]
updated the repo settings: 
Screenshot

**Pull request merged**